src/main/java/cz/cvut/kbss/termit/persistence/dao/skos/SKOSImporter.java

```diff
@@ -101,119 +106,75 @@ private void parseDataFromStreams(String mediaType, InputStream... inputStreams)
+            if (glossary.isIRI()) {
+                this.glossary = (IRI) glossary;
+                return;
+            } else {
+                throw new IllegalArgumentException(
+                    "Blank node skos:ConceptScheme not supported.");
+            }
+        } else {
+            throw new IllegalArgumentException(
+                "No unique skos:ConceptScheme found in the provided data.");
         }
-        throw new IllegalArgumentException(
-                "No vocabulary or ontology found in the provided data. This means target storage context cannot be determined.");
     }
 
-    private Vocabulary constructVocabularyInstance() {
+    private Vocabulary resolveVocabularyFromGlossary(final String vocabularyIri) {
         final Vocabulary instance = new Vocabulary();
         instance.setUri(URI.create(vocabularyIri));
-        final Set<Statement> labels = model.filter(vf.createIRI(vocabularyIri), DCTERMS.TITLE, null);
+
+        final Glossary gls = new Glossary();
+        gls.setUri(URI.create(glossary.stringValue()));
+        instance.setGlossary(gls);
+
+        final cz.cvut.kbss.termit.model.Model mdl = new cz.cvut.kbss.termit.model.Model();
+        mdl.setUri(URI.create(vocabularyIri + "/model"));
+        instance.setModel(mdl);
+
+        final Set<Statement> labels = model.filter(glossary, DCTERMS.TITLE, null);
         labels.stream().filter(s -> {
             assert s.getObject() instanceof Literal;
             return Objects.equals(config.get(ConfigParam.LANGUAGE),
-                    ((Literal) s.getObject()).getLanguage().orElse(config.get(ConfigParam.LANGUAGE)));
+                ((Literal) s.getObject()).getLanguage().orElse(config.get(ConfigParam.LANGUAGE)));
         }).findAny().ifPresent(s -> instance.setLabel(s.getObject().stringValue()));
         return instance;
     }
 
-    private void insertTermGlossaryMembership() {
-        LOG.trace("Ensuring glossary membership statements (skos:inScheme) exist for terms.");
-        final Optional<IRI> glossary = resolveGlossary();
-        if (!glossary.isPresent()) {
-            LOG.warn("No glossary found. Will not be able to ensure terms are in a SKOS scheme.");
-            return;
-        }
-        model.addAll(model.filter(null, RDF.TYPE, SKOS.CONCEPT).stream()
-                          .map(s -> vf.createStatement(s.getSubject(), SKOS.IN_SCHEME, glossary.get()))
-                          .collect(Collectors.toList()));
-    }
-
-    private Optional<IRI> resolveGlossary() {
-        final IRI vocabularyId = vf.createIRI(vocabularyIri);
-        final List<Value> glossary = model
-                .filter(vocabularyId, vf.createIRI(cz.cvut.kbss.termit.util.Vocabulary.s_p_ma_glosar), null).stream()
-                .map(Statement::getObject).collect(Collectors.toList());
-        return glossary.size() != 0 ? Optional.of((IRI) glossary.get(0)) : Optional.empty();
-    }
-
     private void insertTopConceptAssertions() {
```

Comment:
Code formatting in this method is horrible.

Reply:
Code reformatting in Idea does not change anything. I am not aware about any other code style defined.